### ISSUE TITLE
Enhance Dictionary Organization in Settings for Users with Extensive Dictionary Collections

### ISSUE THREAD

**Issue by AdriGrana** [themoeway/yomitan#796]
## Enhance Dictionary Organization in Settings for Users with Extensive Dictionary Collections

Hello Yomitan Team,

I'm a big fan of Yomitan and appreciate the ability to integrate 50+ dictionaries into the program. However, I've encountered a challenge with organizing a large number of dictionaries within the settings. Currently, the only organizational tool available is setting the priority of each dictionary. While this works well for a smaller collection (under 10 dictionaries), it becomes cumbersome and inefficient for users who, like myself, prefer a more extensive library.

### Suggestion for Improvement:

I propose introducing a feature that allows users to categorize dictionaries into different groups. For example, we could have categories for Yojijukugo dictionaries, grammar dictionaries, and general monolingual dictionaries. This could be visually represented in the settings viewer as different tabs for each self-defined category, rather than a single, extensive list.

In the image below you can see what of a mess it currently is to organize them all, they don't even fit on a page without scrolling.

![Screenshot of Current Dictionary Viewer](https://github.com/themoeway/yomitan/assets/63470662/1f761741-ff32-4424-a640-038768c1aeb8)

Alternative approaches, such as a hierarchy of virtual folders with customizable names, could also be considered. The goal is to simplify the management of numerous dictionaries, making it easier to understand their interactions and applications.

Additionally, implementing a feature to set priority at the group/category level would significantly enhance usability. For instance, when encountering a Yojijukugo, having all relevant dictionary definitions prioritized at the top would be ideal, rather than manually adjusting the priority of individual dictionaries within a mixed list.

### Conclusion:

This suggestion aims to refine the dictionary setup process in Yomitan, specifically for users with extensive collections. It seeks to maintain the current functionality in search and definition retrieval while improving the organization and prioritization settings for a better user experience. So this means that in the search page nothing would really change, it's really just to have a better overview on what dictonaries and types of dictonaries are at play.

I'm open to feedback and other ideas on how to achieve this goal and look forward to discussing this further with the community and developers.

**Comment by Maltesaa** [comment]
I think that even just by removing the priority numbers and making each dictionary drag and droppable would be a huge improvement. So then you would just put the dictionaries in the order you want them, top = highest priority, bottom = lowest priority.

I don't know if there's a specific reason for going with priority numbers but I've always felt that to be a bit unintuitive. Making the dictionaries drag and droppable would speed things up so much as well.

**Comment by AdriGrana** [comment]
Hey thanks for commenting on here! The drag and drop sounds interesting especially in combination with relative priorities where it's just highest to lowest (though you have to be able to drag and drop them to anywhere you'd like in the list also change the order of individual dicts at any time). This would certainly solve some issues that I talked mentioned.

But the problem of not getting a clear overview of what types of dictionaries are where in a glance would still persist because you'd still have one huge list which is just really unwieldy to look at and conceptualize mentally what is where. I think categories/tags/folders or whatever of some sort are dearly needed anyways. But in combination with your drag/drop idea and abolishing the priorities entirely it sounds even better I think.

**Comment by Kuuuube** [comment]
This is somewhat added now. Dictionaries will properly order themselves top to bottom when they all have the same priority. And there are now arrows so you can easily shift dicts around a few positions up or down.

**Comment by AdriGrana** [comment]
@Kuuuube  Yeah that's cool but still far from something where you can get a good overview about all the dicts and organize them more efficiently than just a linear approach. For example if I want a 四字熟語 dictonary to always be on top when there is an entry for it, I have to go through all my dicts and fish out all 四字熟語 dictonaries and assign a high priority. Same for other dictonaries that specialize in a certain thing that I only need when whatever they specialize in shows up. Having different categorize just would make it easier to do that and also to keep an overview over it for when I haven't touched the priorities in a while because it would be much easier to see the structure.

Anyways I understand it if it's too much trouble to implement or not really needed for most people, but I think it would enhance the experience for people with many dictonaries a lot.

**Comment by ganqqwerty** [comment]
I think this one is related to my idea here: https://github.com/themoeway/yomitan/issues/568

**Comment by aexiting** [comment]
I'll pick this up. And maybe have a fix for it by next week.
